client/src/lib/error-telemetry.js

```diff
@@ -0,0 +1,68 @@
+/**
+* The error telemetry module handles sending error logging to the backend.
+*
+* NOTE: we _might_ want in the future to have only one Telemetry class and
+* handle both error and "metrics" telemetry, but let's start with a smaller
+* scope at least for now.
+*/
+const Tail   = require('tail').Tail;
+const fs     = require('fs');
+const logger = require('winston');
+
+const TARGET = '/tmp/test';
+
+class ErrorTelemetry {
+  constructor(app, options) {
+    this.app = app;
+    this.options = options;
+  }
+
+  setup() {
+    logger.info('Setting up error logging...');
+    var loggingFile = this.fileToWatch();
+
+
+    // FIXME: this is wrong, we should probably /just/ retry, but handling this an error
+    // BTW, we should define a way to also report critical errors like those to the backend
+    // The logging file being absent, or some critical parts being wrong on the Jenkins instance,
+    // basically nothing should never make the client crash I think
+    // (if a connection is established, sure)
+    if(!fs.existsSync(loggingFile)) {
+      throw new Error('logging file ' + loggingFile + ' not found. ');
+    }
+
+    const tail = new Tail(loggingFile, {
```

Comment:
Would there be any utility in making `tail` an object variable, e.g. `this.tail`? Will future methods need to access this tail object?

Reply:
Possibly, but YAGNI for now IMO. Easy enough to change if so.